piksi_multi_cpp/src/sbp_callback_handler/geotf_handler.cc

```diff
@@ -0,0 +1,71 @@
+#include "piksi_multi_cpp/sbp_callback_handler/geotf_handler.h"
+
+#include <functional>
+
+namespace piksi_multi_cpp {
+namespace s = std::placeholders;
+
+GeoTfHandler::GeoTfHandler(const ros::NodeHandle& nh,
+                           const std::shared_ptr<sbp_state_t>& state)
+    : base_pos_llh_handler_{std::bind(&GeoTfHandler::callbackToBasePosLlh, this,
+                                      s::_1, s::_2),
+                            SBP_MSG_BASE_POS_LLH, state},
+      pos_llh_handler_{
+          std::bind(&GeoTfHandler::callbackToPosLlh, this, s::_1, s::_2),
+          SBP_MSG_POS_LLH, state},
+      nh_(nh) {
+  geotf_.initFromRosParam();
+  if (geotf_.hasFrame("enu")) reset_position_ = ResetEnuOrigin::kNo;
+  geotf_.addFrameByEPSG("ecef", 4978);
+  geotf_.addFrameByEPSG("wgs84", 4326);
```

Comment:
Not that this is likely to happen, but could lead to confusion - what would happen if someone configures some other frame as 4978 as ecef in the rosparams? then this would fail and return weird results.

Maybe we could check return value of addFrameByEPSG (https://github.com/ethz-asl/geodetic_utils/blob/master/geotf/src/geodetic_converter.cc#L90), it gives false if the frame is already defined and output an error?

Reply:
Thanks, good catch!